tests/ui/missing_const_for_fn/cant_be_const.rs

```diff
@@ -181,3 +181,9 @@ mod msrv {
         }
     }
 }
+
+mod with_extern {
+    extern "C-unwind" fn c_unwind() {}
+    extern "system" fn system() {}
+    extern "system-unwind" fn system_unwind() {}
```

Comment:
I'm a little bit unsure should we report them with help messages like "if your are on nightly toolchain, enable `const_extern_fn` feature with `#![feature(const_extern_fn)]`" instead of just skip them, it surely will make some noise tho.

Reply:
If you really want to add that and can detect that a nightly toolchain is a build requirement (for example via a toolchain file), I'm not opposed to it.

Reply:
Oh, that reminds me I probably (???) could use this [`cx.sess().is_nightly_build`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_session/session/struct.Session.html#method.is_nightly_build) to check that but I assume it will be useless in our test cases, since it will be `true` in all of them...

Well, I'll keep this in mind for future reference, maybe we could have some attribute like `#[clippy::toolchain = "nightly"]` in the future, but for now this might be fine (I think)~

Reply:
For the record, the problem with `is_nightly_build` is that it's true when compiled on nightly. This is the wrong criteria, the code might still accept compiling on stable.

Reply:
oh right... I see, that'll made the users having to force nightly toolchain if they only follow the suggestion once...

hmmm, I understand why you mentioned "via a toolchain file" now, as it seems like the only sensible way here. But doing so would require clippy to have some basic de-serializer written to match that file, and I saw there's a security flaw with that file on this [rustup issue](https://github.com/rust-lang/rustup/issues/2879), so I'm expecting `rustup` to make some changes regarding that file, meaning that we'll have to change them too~

For these reasons I think it might not worth the efforts, and lets just call this perfect imperfection :D